main.py

```diff
@@ -16,52 +16,104 @@ def id2path(id):
     return id.split("::")[0]
 
 
-def focal2result(syncer, repos_root, obj):
+def java_workdir_dict(test_ids: list[str]) -> dict[str, list[str]]:
+    """split a list of test ids into a dict of workdir to file path
+    this solves the LSP TimeoutError for JAVA with too much subdirectories
+
+    Args:
+        test_ids (list[str]): [path/to/test/file::test_func_name]
+
+    Returns:
+        dict[str, list[str]]: {workdir: [test_id, ...], ...}
+    """
+    workdir_dict = {}
+    for test_id in test_ids:
+        file_path = id2path(test_id)
+        workdir = file_path.rsplit("test", 1)[0]
+        if workdir not in workdir_dict.keys():
+            workdir_dict[workdir] = []
+        workdir_dict[workdir].append(test_id)
+    return workdir_dict
+
+
+def focal2result(syncer: Synchronizer, repos_root, obj):
     file_path = os.path.join(repos_root, id2path(obj["test_id"]))
     src_lineno, src_col_offset = obj["focal_loc"]
     test_lineno, test_col_offset = obj["test_loc"]
 
-    fake_loc = Location(
-        path2uri(file_path),
-        Range(
-            Position(test_lineno - 1, test_col_offset),
-            Position(test_lineno - 1, test_col_offset + 1),
-        ),
-    )
+    langID = syncer.langID
+
+    # only python ast is 1-indexed, tree-sitter and LSP are 0-indexed
+    match langID:
+        case LANGUAGE_IDENTIFIER.PYTHON:
+            src_lineno -= 1
+            test_lineno -= 1
+        case LANGUAGE_IDENTIFIER.JAVA:
+            # todo: find out why this happens
```

Comment:
on `data/repos/java_example` the `col_offset` does not have to be changed, but needs for `spring-cloud/spring-cloud-netflix`.
I guess this is caused by both tabs and spaces being allowed for JAVA.
I will fix this during the weekend.